decidim-surveys/app/helpers/decidim/surveys/publish_answers_helper.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Surveys
+    module PublishAnswersHelper
+      def question_answer_is_publicable(question_type)
+        ignored_question_types = %w(short_answer long_answer separator files).freeze
+
+        ignored_question_types.exclude?(question_type)
+      end
+
+      # Renders the chart for the given question.
+      # Uses chartkick to render the chart.
+      #
+      # @param question_id [Integer] the question id for Decidim:
+      def chart_for_question(question_id)
+        question = Decidim::Forms::Question.includes(answers: { choices: [:answer_option, :matrix_row] }).find(question_id)
+
+        Chartkick.options = {
+          library: { animation: { easing: "easeOutQuart" } },
+          colors: colors_list
+        }
+
+        case question.question_type
+        when "single_option", "multiple_option"
+          options_column_chart_wrapper(question)
+        when "matrix_single", "matrix_multiple"
+          matrix_stack_chart_wrapper(question)
+        when "sorting"
+          sorting_stack_chart_wrapper(question)
+        else
+          "Unknown question type"
+        end
+      end
+
+      private
+
+      def sorting_stack_chart_wrapper(question)
+        counts = Hash.new { |hash, key| hash[key] = Hash.new(0) }
+
+        question.answers.each do |answer|
+          answer.choices.each do |choice|
+            name = translated_attribute(choice.answer_option.body)
+            row = choice.position + 1
+
+            counts[row][name] += 1
+          end
+        end
+
+        tally = counts.map do |name, row_data|
+          {
+            name:,
+            data: row_data.map { |row, count| [row, count] }
+          }
+        end
+
+        bar_chart(tally.sort_by { |data| data[:name] }, stacked: true, download: true)
+      end
+
+      def matrix_stack_chart_wrapper(question)
+        counts = Hash.new { |hash, key| hash[key] = Hash.new(0) }
+
+        question.answers.each do |answer|
+          answer.choices.each do |choice|
+            name = translated_attribute(choice.answer_option.body)
+            row = translated_attribute(choice.matrix_row.body)
+
+            counts[name][row] += 1
+          end
+        end
+
+        tally = counts.map do |name, row_data|
+          {
+            name:,
+            data: row_data.map { |row, count| [row, count] }
+          }
+        end
+
+        column_chart(tally, stacked: true, legend: :right, download: true)
+      end
+
+      def options_column_chart_wrapper(question)
+        tally = question.answers.map { |answer| answer.choices.map { |choice| translated_attribute(choice.answer_option.body) } }.tally
+
+        column_chart(tally, download: true)
+      end
+
+      def colors_list
```

Comment:
To be honest, i would put this in a initializer. There may be instances that may want to customize on accessibility / contrast ratio reasons.

Reply:
I'd prefer to not add another configuration option "just in case". If any instance needs to change this because of a11y, then we should fix it in Decidim vanilla, as we should aim to have good accessibility support built-in by default.  

Mind that these colors were based on what ChartJS offers, so they should be pretty safe to use.